Apps/W1/Sustainability/app/src/Journal/SustainabilityJnlLine.Table.al

```diff
@@ -334,15 +334,30 @@ table 6214 "Sustainability Jnl. Line"
         Validate("Document No.", SustainabilityJournalMgt.GetDocumentNo(IsPreviousLineValid, SustainabilityJnlBatch, PreviousLine."Document No.", "Posting Date"));
     end;
 
-    local procedure GetDefaultDimensionsFromAccount()
+    procedure CreateDimFromDefaultDim(FieldNo: Integer)
     var
         DefaultDimSource: List of [Dictionary of [Integer, Code[20]]];
+    begin
+        InitDefaultDimensionSources(DefaultDimSource, FieldNo);
+        CreateDim(DefaultDimSource);
+    end;
+
+    local procedure InitDefaultDimensionSources(var DefaultDimSource: List of [Dictionary of [Integer, Code[20]]]; FieldNo: Integer)
+    begin
+        DimMgt.AddDimSource(DefaultDimSource, Database::"Sustainability Account", "Account No.", FieldNo = Rec.FieldNo("Account No."));
+        OnAfterInitDefaultDimensionSources(Rec, DefaultDimSource, FieldNo);
+    end;
+
+    procedure CreateDim(DefaultDimSource: List of [Dictionary of [Integer, Code[20]]])
+    var
+        OldDimSetID: Integer;
     begin
         "Shortcut Dimension 1 Code" := '';
         "Shortcut Dimension 2 Code" := '';
+        OldDimSetID := "Dimension Set ID";
+        "Dimension Set ID" := DimMgt.GetRecDefaultDimID(Rec, CurrFieldNo, DefaultDimSource, "Source Code", "Shortcut Dimension 1 Code", "Shortcut Dimension 2 Code", 0, 0);
 
-        DimMgt.AddDimSource(DefaultDimSource, Database::"Sustainability Account", "Account No.", true);
-        Validate("Dimension Set ID", DimMgt.GetRecDefaultDimID(Rec, CurrFieldNo, DefaultDimSource, "Source Code", "Shortcut Dimension 1 Code", "Shortcut Dimension 2 Code", 0, 0));
```

Comment:
Look into the `DimensionManagement.Codeunit.al` and found the procedure:

```
procedure GetRecDefaultDimID(RecVariant: Variant; CurrFieldNo: Integer; DefaultDimSource: List of [Dictionary of [Integer, Code[20]]]; SourceCode: Code[20]; var GlobalDim1Code: Code[20]; var GlobalDim2Code: Code[20]; InheritFromDimSetID: Integer; InheritFromTableNo: Integer) Result: Integer
    var
        DefaultDimID: Integer;
        IsHandled: Boolean;
    begin
        IsHandled := false;
        OnBeforeGetRecDefaultDimID(RecVariant, CurrFieldNo, DefaultDimSource, SourceCode, InheritFromDimSetID, InheritFromTableNo, Result, IsHandled);
        if IsHandled then
            exit(Result);
        DefaultDimID := GetDefaultDimID(DefaultDimSource, SourceCode, GlobalDim1Code, GlobalDim2Code, InheritFromDimSetID, InheritFromTableNo);
        OnAfterGetRecDefaultDimIDProcedure(RecVariant, CurrFieldNo, DefaultDimSource, SourceCode, InheritFromDimSetID, InheritFromTableNo, GlobalDim1Code, GlobalDim2Code, DefaultDimID);
        exit(DefaultDimID);
    end;
```

Can we utilize some of the events in the above procedure instead of creating the new `OnAfterCreateDim`? We can identify the context with the help of the `RecVariant` variable.

Reply:
I'm no big fan of Variants as outlinede above but looking into the event `OnAfterGetRecDefaultDimIDProcedure` I think we should have all required parameters:
- RecVariant -> Provides Information about the related table + we can get the OldDimSetId
- GlobalDim1Code, GlobalDim2Code, DefaultDimID are the required paramters to overwrite the dimensionSet id afterswards.